ui/app/components/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -0,0 +1,375 @@
+    ethTransactionAmount: PropTypes.string,
+    ethTransactionFee: PropTypes.string,
+    ethTransactionTotal: PropTypes.string,
+    fiatTransactionAmount: PropTypes.string,
+    fiatTransactionFee: PropTypes.string,
+    fiatTransactionTotal: PropTypes.string,
+    hexGasTotal: PropTypes.string,
+    balance: PropTypes.string,
+    currentCurrency: PropTypes.string,
+    conversionRate: PropTypes.number,
+    clearConfirmTransaction: PropTypes.func,
+    cancelTransaction: PropTypes.func,
+    clearSend: PropTypes.func,
+    sendTransaction: PropTypes.func,
+    editTransaction: PropTypes.func,
+    showCustomizeGasModal: PropTypes.func,
+    updateGasAndCalculate: PropTypes.func,
+    showTransactionConfirmedModal: PropTypes.func,
+    // Component props
+    action: PropTypes.string,
+    hideDetails: PropTypes.bool,
+    hideData: PropTypes.bool,
+    detailsComponent: PropTypes.node,
+    dataComponent: PropTypes.node,
+    summaryComponent: PropTypes.node,
+    contentComponent: PropTypes.node,
+    title: PropTypes.string,
+    subtitle: PropTypes.string,
+    hideSubtitle: PropTypes.bool,
+    valid: PropTypes.bool,
+    errorMessage: PropTypes.string,
+    warning: PropTypes.string,
+    identiconAddress: PropTypes.string,
+    onEdit: PropTypes.func,
+    onEditGas: PropTypes.func,
+    onCancel: PropTypes.func,
+    onSubmit: PropTypes.func,
+  }
+
+  componentDidUpdate () {
+    const {
+      transactionStatus,
+      showTransactionConfirmedModal,
+      history,
+      clearConfirmTransaction,
+    } = this.props
+
+    if (transactionStatus === 'dropped') {
+      showTransactionConfirmedModal({
+        onHide: () => {
+          clearConfirmTransaction()
+          history.push(DEFAULT_ROUTE)
+        },
+      })
+
+      return
+    }
+  }
+
+  getError () {
+    const INSUFFICIENT_FUNDS_ERROR = this.context.t('insufficientFunds')
+    const TRANSACTION_ERROR = this.context.t('transactionError')
+    const {
+      balance,
+      conversionRate,
+      hexGasTotal,
+      txData: {
+        simulationFails,
+        txParams: {
+          value: amount,
+        } = {},
+      } = {},
+    } = this.props
+
+    const insufficientBalance = balance && !isBalanceSufficient({
+      amount,
+      gasTotal: hexGasTotal || '0x0',
+      balance,
+      conversionRate,
+    })
+
+    if (insufficientBalance) {
+      return {
+        valid: false,
+        errorMessage: INSUFFICIENT_FUNDS_ERROR,
+      }
+    }
+
+    if (simulationFails) {
+      return {
+        valid: false,
+        errorMessage: TRANSACTION_ERROR,
+      }
+    }
+
+    return {
+      valid: true,
+    }
+  }
+
+  validateEditGas ({ gasLimit, gasPrice }) {
+    const { t } = this.context
+    const {
+      balance,
+      conversionRate,
+      txData: {
+        txParams: {
+          value: amount,
+        } = {},
+      } = {},
+    } = this.props
+
+    const INSUFFICIENT_FUNDS_ERROR = t('insufficientFunds')
```

Comment:
I think it might be better to store errors as their translation keys, as opposed to their translations. And then translate them at the point of render.

Reasons:
(1) Data stored will not be a 'view' value, just data that represents the current state of errors
(2) Debugging will be easier if the users state logs are all in english, as opposed to some parts of state that representing error handling being in a different laguage

Also, perhaps these errors should be moved to a constants file